node/src/service.rs

```diff
@@ -0,0 +1,54 @@
+//! background service
+
+use crate::eth::{backend, miner::MiningMode, pool::Pool};
+use std::{
+    future::Future,
+    pin::Pin,
+    sync::Arc,
+    task::{Context, Poll},
+};
+use tracing::trace;
+
+/// The type that drives the blockchain's state
+///
+/// This service is basically an endless future that continuously polls the miner which returns
+/// transactions for the next block, then those transactions are handed off to the
+/// [backend](backend::mem::Backend) to construct a new block, if all transactions were successfully
+/// included in a new block they get purged from the `Pool`.
+pub struct NodeService {
+    /// the pool that holds all transactions
+    pool: Arc<Pool>,
+    /// holds the blockchain's state
+    backend: Arc<backend::mem::Backend>,
+    /// the miner responsible to select transactions from the `pool´
+    miner: MiningMode,
```

Comment:
Brainstorming, wonder if we can make this a bit more abstract, so we can refactor these to be "merge-ready", is the miner just an entity that pops txs from the pool?

Reply:
yes, a miner gets access to the pool and determines what transactions to mine